### PR TITLE
Revival update

### LINKED ISSUE #1: [FEATURE] Meaningful Space: Add the option to fix elements
This is awesome, but It's halfway there.
If we could drag and lock certain nodes in space, we would have the first real map of content, not the dynamic diagram the graph currently is.

imagen
As cool as Obsidian can distribute nodes, we lack the ability to give space meaning.

### LINKED ISSUE #2: [FEATURE] - in Group filters, turn filtered dots into an Emojii - instead of a Color
.

### LINKED ISSUE #3: [FEATURE] - Small tiny 1KB or less custom icons for every node , fetched from YAML inside of note.
.

### LINKED ISSUE #6: [BUG] - Position isn't restored upon loading an app if graph isn't in the leftmost pane
If I have multiple panes, one of them is the graph, then if I close Obsidian and open it back again, position is restored upon loading only if the graph is in the leftmost pane or in is the only pane (which is treated as leftmost, I assume), but if the graph is in any pane other then the left one - the restoring doesn't occur upon loading the app. Restoring still works after loading if I use a shortcut or reload the graph or open a new pane. It's just about the first loading of the app.

### LINKED ISSUE #8: Add save/restore buttons on graph view leaf
Instead of requiring the use of commands, add save/restore buttons alongside "Close" and "More options".

Method suggested by koala:
>I think you'd be better off using addAction on the graph view leaf https://github.com/obsidianmd/obsidian-api/blob/d4b79f95ccf3838a305915a137a844250766d023/obsidian.d.ts#L1514 and storing a reference to each added button on your plugin. Then you can just call HTMLElement.detach() to remove it.

### LINKED ISSUE #12: Automatic Position Saving?
Hey all

Thanks very much for making this. I use the graph as my exclusive browsing tool and having positions saved allows me to learn my network visually. 

One question, any suggestions on how I might get this plugin to save every 30 secs, or save every time I close obsidian?

### LINKED ISSUE #13: manifest still has version 0.1.5 and the plugin  appears as version 0.1.5 in obsidian's plugin search
should change version to 0.2

### LINKED ISSUE #14: This plugin is not compatible with Obsidian v.1.0.3
Failed to load plugin persistent-graph

### LINKED ISSUE #15: Graph locks after restoring node positions
Hello, I am using persistant graph with the latest release (v1.1.9). I am seeing an issue where, when node positions are restored, they become locked and cannot be moved by hand. When "Automatically restore node positions" is off, I can open a new graph view and move up to ~5 nodes, but then they all become locked in place (I cannot drag them). Before nodes become locked, all commands work as expected (save/reload, run/stop sim), although the "save" command causes the nodes to become locked immediately. When  "Automatically restore node positions" is on, node positions are always fixed in the last saved layout, and I can't move anything. 

I saw this issue in v1.0.3 and thought updating would help, but it did not.

### LINKED ISSUE #16: [FEATURE] Notification that save was successful
As the title says, I would like to have a notification to feedback to me that my save was successful. I currently press my hotkey combination to save, and it does work, but I have no way to know if it actually saved it or not

### LINKED ISSUE #18: Lost structure after renaming a folder
I'm displaying just one folder in my graph – my framework. After I've renamed this one, and updated the filter path of the graph, I tried to restore the saved position; but it was vanished.

### LINKED ISSUE #19: [Feature request] Add buttons to save and restore node positions
Add buttons next to those that already exist in Graph View interface to save and restore node positions.

This would facilitate the use of the plugin because it would expedite the use of its features.

### LINKED ISSUE #20: Add new note using graph view
I love this plugin 👏 👏 

How can you create a new note from a click event inside the graph view ?

### LINKED ISSUE #21: Move several nodes at once
How can you move several nodes positions by selecting all ?

### LINKED ISSUE #22: Feature request: stop running automatically
The command to start running the graph simulation should stop itself after a certain point, probably when the nodes stop moving significantly.

PS No activity here lately beyond issues. Dead project?

### LINKED ISSUE #23: [Feature Request] - Automatically run graph simulation
Would be cool to have a toggle to auto-run the graph sim every time it opens.

### LINKED ISSUE #24: PlugIn don't working
I tried to reinstall and also to restart Obsidian, nothing worked. I am in the Obsidian version 1.8.7.

### LINKED ISSUE #25: Plugin is not working
I installed today, using obsidian v 1.8.7. Plugin simply does not do what it was supposed to do.

### LINKED ISSUE #26: I've been trying to use the plugin for 2 months without success.
1 - The automatic save doesn't work, so I always have to save the position manually. But that's not the worst part, in fact I had already gotten used to it.

2 -  The worst problem is that suddenly all the positions are unpinned and I lose all the previous work of organizing nodes. I tried to understand what exactly I do before this happens, but I can't find any pattern. 

3 - Another thing that happens is that whenever I change the name of a title the node moves from its place. Can someone help me with this, please :)

### LINKED ISSUE #27: Excalidraw plugin ruining the position of grap notes
Excalidraw plugin ruining the position of grap notes. Every time i've installed the excalidraw, then i see the graph.json and data.json was updated and then the graph was ruined. I fixed it with restore the old data and, i don't know how, save and restore button

### LINKED ISSUE #29: When you click "save graph node position" not in the "graph view", you will broken your saves
When you click "save graph node position" not in the "graph view", you will broken your saves. I've noticed it in canvas, when i by chance press "the graph save".

How to fix that: recovery "/persistence graph/data.json" from sync of whatever you use for recovery

### PR DESCRIPTION
**Pull request by Gyrandola**
In this PR I revisit this old plugin to give it some love, fix issues and potentially add some of the requested features.
I'm aware it will not be atomic at all, hope it wont be too much of a problem. Can split if needed.

## Bugs
- [x] Fixes #6 
- [x] Fixes #13 
- [x] Fixes #14 
- [x] Fixes  #15
(not replicable)
- [x] Fixes #18 
- [x] Fixes #24 
(not replicable)
- [x] Fixes #25 
(not replicable)
- [x] Fixes #26
(linked with issue 12)
- [x] Fixes #27 
(~~might not be fixable if it's a compatibility issue, will look into it~~. Found no issues on my end)
- [x] Fixes #29 

## Features
- [x] Resolves #1
(reluctant to add this but I will)
- [x] Resolves #8 
- [x] Resolves #12 
- [x] Resolves #16 and closes #17 
- [x] Resolves #19 
- [x] Resolves #22 
( I am not sure what to make of this. I have found little use for the simulation commands as they cause my graph to jitter senselessly)
image

## Beyond project scope
Most of these are a bit beyond the original scope of this plugin, I feel. Besides, they are already implemented by many others and might conflict or cause bloat.
- [x] Closes #2 
- [x] Closes #3 
- [x] Closes  #20 
- [x] Closes  #21
- [x] Closes #23 
- [x] Closes  #20

### CODE REVIEW

**Comment by sens0rer** [comment]
This is really cool! Can you share the built file as well?

**Comment by Sanqui** [comment]
The code looks good and the functionality I tested locally behaves as expected.  Thank you for your contribution, I'm about to ship this as 0.3.0!
